addons/event_crm/models/event_crm.py

```diff
@@ -0,0 +1,111 @@
+    lead_user_id = fields.Many2one('res.users', string='Salesperson', help="Link a salesperson to the lead that will be created.")
+    lead_tag_ids = fields.Many2many('crm.lead.tag', string='Tags', help="Link tags to the lead that will be created.")
+
+    def _create_leads(self, registration_ids):
+        """
+        Create leads based on the type of lead creation of the rule and apply to this leads
+        some prefilled values based on the rule and the registration.
+        """
+        for rule in self:
+            vals_list = []
+            registrations = registration_ids
+            if rule.registration_filter_domain != '[]':
+                registrations = registration_ids.search([('id', 'in', registration_ids.ids)] + safe_eval(rule.registration_filter_domain))
+            if rule.active and registrations and rule._check_rule(registrations):
+                lead_values = {
+                    'user_id': rule.lead_user_id.id,
+                    'type': rule.lead_type,
+                    'team_id': rule.lead_sales_team_id.id,
+                    'tag_ids': rule.lead_tag_ids,
+                    'registration_rule_id': rule.id,
+                    'referred': 'event',
+                }
+                if rule.lead_creation_basis == 'attendee':
+                    for registration in registrations:
+                        values = lead_values.copy()
+                        values.update(registration._get_registration_lead_values(rule.lead_creation_basis))
+                        vals_list.append(values)
+                else:
+                    if len(registrations) > 1 or not registrations[0].main_registration_id:
+                        lead_values.update(registrations._get_registration_lead_values(rule.lead_creation_basis))
+                        vals_list.append(lead_values)
+                    else:
+                        for lead in registrations.main_registration_id.lead_ids:
+                            if lead.description:
+                                lead.description += registrations._get_lead_description()
+                            registrations.lead_ids += lead
+
+                if vals_list:
+                    self.sudo().env['crm.lead'].create(vals_list)
+
+    def _check_rule(self, registration_ids):
+        """
+        Check if the field of the registrations match with the rule's conditions.
+        """
+        if not self.company_id or self.company_id == registration_ids.company_id:
+            if (
+                (not self.event_type_ids and not self.event_id) or
+                (registration_ids.event_id.event_type_id in self.event_type_ids and not self) or
+                (not self.event_type_ids and self.event_id == registration_ids.event_id) or
+                (registration_ids.event_id.event_type_id in self.event_type_ids or self.event_id == registration_ids.event_id)
```

Comment:
maybe extract some condition parts to variable to ease reading.
btw, this condition will never be hit:
`registration_ids.event_id.event_type_id in self.event_type_ids and not self` as is not self, self.event_type_ids will be null.
It will give:
```
no_rule_event_type = not self.event_type_ids
event_type_matches = registration_ids.event_id.event_type_id in self.event_type_ids
event_matches = self.event_id == registration_ids.event_id

if (
    (no_rule_event_type and not self.event_id) or
    (no_rule_event_type and event_matches) or
    event_type_matches or 
    event_matches
)
```
that could be simplified in 
```
if event_type_matches or event_matches or (no_rule_event_type and not self.event_id):
```
because `if no_event_type and event_matches` is the same then `if event_matches`.

For `no_rule_event_type and not self.event_id`: is it possible to apply a rule of no event type and no event is set on the rule ? with domain maybe ?